streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -213,6 +212,31 @@ public <K, V> void send(final String topic,
                     keyClass,
                     valueClass),
                 exception);
+        } catch (final SerializationException exception) {
+            final ProducerRecord<K, V> record = new ProducerRecord<>(topic, partition, timestamp, key, value, headers);
+            final ProductionExceptionHandler.ProductionExceptionHandlerResponse response;
+            try {
+                response = productionExceptionHandler.handleSerializationException(record, exception);
+            } catch (final Exception fatalUserException) {
+                log.error("Serialization error callback failed after serialization error for record {}", record, fatalUserException);
+                final ProducerRecord<byte[], byte[]> rec = new ProducerRecord<>(topic, partition, timestamp, keyBytes, valBytes, headers);
```

Comment:
valBytes is always null - maybe it is more straightforward to supply a null here?